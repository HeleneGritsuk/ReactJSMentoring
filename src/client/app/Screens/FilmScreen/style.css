.homeBtn {
  margin-left: auto;
  margin-left: auto;
  background: white;
  color: #ff8178;
  text-decoration: none;
  padding: 10px;
  border-radius: 5px;
}

.filmInfo {
  display: flex;
  padding: 0 40px 50px;
  align-items: flex-start;
}
.filmInfo__img {
  width: 300px;
  height: auto;
}
.filmInfo__text {
  margin-left: 30px;
}
