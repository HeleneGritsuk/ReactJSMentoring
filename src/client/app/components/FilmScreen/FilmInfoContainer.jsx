import React from 'react';
import { render } from 'react-dom';
import Header  from '../Header/Header.jsx';
import FilmInfo  from './FilmInfo.jsx';

class FilmInfoContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      filmImg: 'http://www.imfdb.org/images/thumb/6/65/Killbillposter.jpg/300px-Killbillposter.jpg',
      filmTitle: 'Kill Bill',
      genre: 'oscar-winning movies',
      year: '1994',
      duration: 54,
      description: 'Kill Bill: Volume 1 is a 2003 American martial arts film written and directed by Quentin Tarantino. It stars Uma Thurman as the Bride, who swears revenge on a team of assassins (Lucy Liu, Michael Madsen, Daryl Hannah, and Vivica A. Fox) and their leader Bill (David Carradine) after they try to kill her and her unborn child. Her journey takes her to Japan, where she battles the Tokyo yakuza.',
      films: [
        {
          name: 'Kill Bill',
          imgSrc: 'http://www.imfdb.org/images/thumb/6/65/Killbillposter.jpg/300px-Killbillposter.jpg',
          genre: 'drama',
          year: 2014,
          id: 1
        },
        {
          name: 'Kill Bill2',
          imgSrc: 'http://www.imfdb.org/images/thumb/6/65/Killbillposter.jpg/300px-Killbillposter.jpg',
          genre: 'adventure',
          year: 2017,
          id: 2
        }
      ]
    }
  }
  render () {
    return(
      <div className="search">
        <div className="container">
          <Header isFilmScreen = {true} goToHomePage = {this.props.goToHomePage}/>
          <FilmInfo filmInfo = {this.state}></FilmInfo>
        </div>

      </div>
    );
  }
}

export default FilmInfoContainer;
