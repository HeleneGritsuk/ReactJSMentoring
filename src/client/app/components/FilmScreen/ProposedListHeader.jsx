import React from 'react';

const ProposedListHeader = ({genre}) => (
  <div className='searchResults__header'>
    Films by {genre}
  </div>

);

export default ProposedListHeader;
