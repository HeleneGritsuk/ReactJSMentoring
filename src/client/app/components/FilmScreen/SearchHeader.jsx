import React from 'react';

const SearchHeader = ({resultsNumber}) => (
  <div className='searchResults__header'>
    <div className='searchResults__foundNumber'>{resultsNumber} movies found</div>
    <div className='searchResults__sorting'>Sort by <button> release date</button> <button> rating</button></div>
  </div>

);

export default SearchHeader;
