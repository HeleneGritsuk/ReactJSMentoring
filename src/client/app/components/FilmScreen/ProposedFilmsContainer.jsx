import React from 'react';
import { render } from 'react-dom';
import Footer  from '../Footer/Footer.jsx';
import ProposedListHeader  from './ProposedListHeader.jsx';
import ResultsList  from '../ResultsList/ResultsList.jsx';

class ProposedFilmsContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      genre: 'oscar-winning movies',
      films: [
        {
          name: 'Kill Bill',
          imgSrc: 'http://www.imfdb.org/images/thumb/6/65/Killbillposter.jpg/300px-Killbillposter.jpg',
          genre: 'drama',
          year: 2014,
          id: 1
        },
        {
          name: 'Kill Bill2',
          imgSrc: 'http://www.imfdb.org/images/thumb/6/65/Killbillposter.jpg/300px-Killbillposter.jpg',
          genre: 'adventure',
          year: 2017,
          id: 2
        }
      ]
    }
  }
  render () {
    return(
      <section className="searchResults">
        <div className="container">
          <ProposedListHeader genre={this.state.genre}/>
          <ResultsList films={this.state.films}/>
        </div>
      </section>
    );
  }
}

export default ProposedFilmsContainer;
