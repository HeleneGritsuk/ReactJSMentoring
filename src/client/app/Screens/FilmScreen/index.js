import React from 'react';
import { render } from 'react-dom';
import Footer  from '../../components/Footer';
import ProposedFilmsContainer from './components/ProposedFilmsContainer.jsx';
import FilmInfoContainer from './components/FilmInfoContainer.jsx';
import { filmsMockArray } from '../../mocks';
import './style.css';

class FilmScreen extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      genre: 'oscar-winning movies',
      filmImg: 'http://www.imfdb.org/images/thumb/6/65/Killbillposter.jpg/300px-Killbillposter.jpg',
      filmTitle: 'Kill Bill',
      genre: 'oscar-winning movies',
      year: '1994',
      duration: 54,
      description: 'Kill Bill: Volume 1 is a 2003 American martial arts film written and directed by Quentin Tarantino. It stars Uma Thurman as the Bride, who swears revenge on a team of assassins (Lucy Liu, Michael Madsen, Daryl Hannah, and Vivica A. Fox) and their leader Bill (David Carradine) after they try to kill her and her unborn child. Her journey takes her to Japan, where she battles the Tokyo yakuza.',
      films: filmsMockArray
    }
  }

  render () {
    return(
      <div className='main-wrapper filmPage'>
        <FilmInfoContainer goToHomePage = {this.props.goToHomePage} filmInfo = {this.state}/>
        <ProposedFilmsContainer genre = {this.state.genre} films={this.state.films}/>
        <Footer/>
      </div>
    );
  }
}

export default FilmScreen;
