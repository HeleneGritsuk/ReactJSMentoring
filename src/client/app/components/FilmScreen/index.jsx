import React from 'react';
import { render } from 'react-dom';
import Footer  from '../Footer/Footer.jsx';
import ProposedFilmsContainer from './ProposedFilmsContainer.jsx';
import FilmInfoContainer from './FilmInfoContainer.jsx';

class FilmScreen extends React.Component {
  constructor(props) {
    super(props);
  }

  render () {
    return(
      <div className='main-wrapper'>
        <FilmInfoContainer goToHomePage = {this.props.goToHomePage}/>
        <ProposedFilmsContainer/>
        <Footer/>
      </div>
    );
  }
}

export default FilmScreen;
