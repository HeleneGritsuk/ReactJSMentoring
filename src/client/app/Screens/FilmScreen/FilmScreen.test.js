import React from 'react';
import FilmScreen from './index';
import renderer from 'react-test-renderer';


test('FilmScreen should render film view correctly', () => {
  const component = renderer.create(
    <FilmScreen/>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
