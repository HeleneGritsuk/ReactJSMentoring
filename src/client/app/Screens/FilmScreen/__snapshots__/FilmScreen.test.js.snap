// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmScreen should render film view correctly 1`] = `
<div
  className="main-wrapper filmPage"
>
  <div
    className="search"
  >
    <div
      className="container"
    >
      <header>
        <h1>
          netflixroulette
        </h1>
        <a
          className="homeBtn"
          href="#"
          onClick={undefined}
        >
          Search
        </a>
      </header>
      <div
        className="filmInfo"
      >
        <img
          src="http://www.imfdb.org/images/thumb/6/65/Killbillposter.jpg/300px-Killbillposter.jpg"
        />
        <div
          className="filmInfo__text"
        >
          <h2>
            Kill Bill
          </h2>
          <div
            className="filmInfo__genre"
          >
            oscar-winning movies
          </div>
          <div>
            <div>
              1994
            </div>
            <div>
              54
               min
            </div>
          </div>
          <p>
            Kill Bill: Volume 1 is a 2003 American martial arts film written and directed by Quentin Tarantino. It stars Uma Thurman as the Bride, who swears revenge on a team of assassins (Lucy Liu, Michael Madsen, Daryl Hannah, and Vivica A. Fox) and their leader Bill (David Carradine) after they try to kill her and her unborn child. Her journey takes her to Japan, where she battles the Tokyo yakuza.
          </p>
        </div>
      </div>
    </div>
  </div>
  <section
    className="searchResults"
  >
    <div
      className="container"
    >
      <div
        className="searchResults__header"
      >
        Films by 
        oscar-winning movies
      </div>
      <div
        className="searchResults__content"
      >
        <div
          className="filmItem"
        >
          <img
            src="http://www.imfdb.org/images/thumb/6/65/Killbillposter.jpg/300px-Killbillposter.jpg"
          />
          <div
            className="filmItem__info"
          >
            <div
              className="filmItem__title"
            >
              Kill Bill
            </div>
            <div
              className="filmItem__year"
            >
              2014
            </div>
          </div>
          <div
            className="filmItem__genre"
          >
            drama
          </div>
        </div>
        <div
          className="filmItem"
        >
          <img
            src="http://www.imfdb.org/images/thumb/6/65/Killbillposter.jpg/300px-Killbillposter.jpg"
          />
          <div
            className="filmItem__info"
          >
            <div
              className="filmItem__title"
            >
              Kill Bill2
            </div>
            <div
              className="filmItem__year"
            >
              2017
            </div>
          </div>
          <div
            className="filmItem__genre"
          >
            adventure
          </div>
        </div>
      </div>
    </div>
  </section>
  <footer>
    <div
      className="container"
    >
      <p>
        netflixroulette
      </p>
    </div>
  </footer>
</div>
`;
